### PR TITLE
[Enhancement] remove unnecessary starlet dependencies in docker image

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docker/dockerfiles/dev-env/dev-env.Dockerfile [hunk 1]
```diff
@@ -68,3 +68,16 @@ COPY --from=builder $STARROCKS_THIRDPARTY $STARROCKS_THIRDPARTY
...
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/aws
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/boost
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/librocksdb.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libbrpc.a
+RUN rm -r $STARLET_INSTALL_DIR/third_party/lib/aws*
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libaws*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libboost*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libprotobuf*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libprotoc.a
+RUN rm $STARLET_INSTALL_DIR/third_party/bin/protoc*
```
Comment (kevincai) [inline docker/dockerfiles/dev-env/dev-env.Dockerfile:83]: make the deleted items into a file `starlet_redundant_lists.txt`

```
RUN for line in `cat starlet_redundant_lists.txt` ; do rm $STARLET_INSTALL_DIR/$line ; done
```

Also need to create a separate stage to perform the delete operations, and then copy the remainder to $STARLET_INSTALL_DIR

File: docker/dockerfiles/dev-env/dev-env.Dockerfile [hunk 2]
```diff
@@ -54,6 +54,11 @@ FROM starrocks/starlet-artifacts-ubuntu22:${starlet_tag} as starlet-ubuntu
 FROM starrocks/starlet-artifacts-centos7:${starlet_tag} as starlet-centos7
 # determine which artifacts to use
 FROM starlet-${distro} as starlet
+# remove unnecessary and big starlet dependencies
+COPY --from=builder /root/starrocks/docker/dockerfiles/dev-env/starlet_exclude .
+RUN while read line; do \
```
Comment (kevincai) [inline docker/dockerfiles/dev-env/dev-env.Dockerfile:59]: RUN `cd release && cat starlet_exclude | xargs -r rm -rvf`

File: docker/dockerfiles/dev-env/starlet_exclude
Comment (kevincai) [inline docker/dockerfiles/dev-env/starlet_exclude:1]: starlet_exclude_filelist

File: docker/dockerfiles/dev-env/dev-env.Dockerfile (hunk 2 above)
Comment (kevincai) [reply docker/dockerfiles/dev-env/dev-env.Dockerfile:59]: possibly exclude lines start with `#` so it can be easily unmask the delete line.

**Review by kevincai — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**